### PR TITLE
Create UseLocalCompiler.Directory.Build.props

### PR DESCRIPTION
**Pull request by vzarytovskii**
Added props file, which will point to locally build compiler (Debug configuration), which can simply be included in contributor's project (via `<Import Project="UseLocalCompiler.Directory.Build.props"/>`) to test changes easier.

### CODE REVIEW

File: UseLocalCompiler.Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<Project>
+  <PropertyGroup>
+    <LocalFSharpCompilerConfiguration>Debug</LocalFSharpCompilerConfiguration>
```
Comment (T-Gro) [inline UseLocalCompiler.Directory.Build.props:3]:
```suggestion
    <LocalFSharpCompilerConfiguration Condition="'$(LocalFSharpCompilerConfiguration )' != ''">Debug</LocalFSharpCompilerConfiguration>
```

File: UseLocalCompiler.Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    <LocalFSharpCompilerPath>$(MSBuildThisFileDirectory)</LocalFSharpCompilerPath>
```
Comment (T-Gro) [inline UseLocalCompiler.Directory.Build.props:5]:
```suggestion
    <LocalFSharpCompilerPath  Condition="'$(LocalFSharpCompilerPath  )' != ''">$(MSBuildThisFileDirectory)</LocalFSharpCompilerPath>
```

File: UseLocalCompiler.Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    <DisableAutoSetFscCompilerPath>true</DisableAutoSetFscCompilerPath>
+
+    <DotnetFscCompilerPath>$(LocalFSCDllPath)</DotnetFscCompilerPath>
+    <Fsc_DotNET_DotnetFscCompilerPath>$(LocalFSCDllPath)</Fsc_DotNET_DotnetFscCompilerPath>
+
+    <FSharpPreferNetFrameworkTools>False</FSharpPreferNetFrameworkTools>
+    <FSharpPrefer64BitTools>True</FSharpPrefer64BitTools>
+
+    <LocalFSharpBuildBinPath>$(LocalFSharpCompilerPath)/artifacts/bin/fsc/$(Configuration)/net8.0</LocalFSharpBuildBinPath>
```
Comment (T-Gro) [inline UseLocalCompiler.Directory.Build.props:18]:
```suggestion
    <LocalFSharpBuildBinPath>$(LocalFSharpCompilerPath)/artifacts/bin/fsc/$(LocalFSharpCompilerConfiguration)/net8.0</LocalFSharpBuildBinPath>
```

File: UseLocalCompiler.Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <FSharpBuildAssemblyFile>$(LocalFSharpBuildBinPath)/FSharp.Build.dll</FSharpBuildAssemblyFile>
+    <FSharpTargetsPath>$(LocalFSharpBuildBinPath)/Microsoft.FSharp.Targets</FSharpTargetsPath>
+    <FSharpPropsShim>$(LocalFSharpBuildBinPath)/Microsoft.FSharp.NetSdk.props</FSharpPropsShim>
+    <FSharpTargetsShim>$(LocalFSharpBuildBinPath)/Microsoft.FSharp.NetSdk.targets</FSharpTargetsShim>
+    <FSharpOverridesTargetsShim>$(LocalFSharpBuildBinPath)/Microsoft.FSharp.Overrides.NetSdk.targets</FSharpOverridesTargetsShim>
+  </PropertyGroup>
+
+  <UsingTask TaskName="FSharpEmbedResourceText" AssemblyFile="$(FSharpBuildAssemblyFile)" Override="true" />
+  <UsingTask TaskName="FSharpEmbedResXSource" AssemblyFile="$(FSharpBuildAssemblyFile)" Override="true" />
```
Comment (T-Gro) [inline UseLocalCompiler.Directory.Build.props:27]: This has to go into .targets , I am afraid it will fail or not do anything if inside .props

**Comment by psfinaki** [comment]
This should be somehow mentioned in the docs, and also explained how's this better than or different from this tip: https://github.com/dotnet/fsharp/blob/main/DEVGUIDE.md#using-your-custom-compiler-to-build-other-projects

File: UseLocalCompiler.Directory.Build.props (hunk 4 above)
Comment (vzarytovskii) [reply UseLocalCompiler.Directory.Build.props:25]: It seems to be working

**Review by edgarfgp — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Comment by vzarytovskii** [comment]
Gonna merge it as is and add docs if needed

**Comment by edgarfgp** [comment]
Thanks

**Pull request merged**